### ISSUE TITLE
include/exclude option excludes glob imports

### ISSUE THREAD

**Issue by stropho** [kei-ito/rollup-plugin-glob-import#67]
Hi,
I'm wondering about the include/exclude option. The filter checks importee rather than importer. That means
```js
// src/index.js
export * from './subfolder/!(*.test).js'
```
and the filter is applied for the string `'./subfolder/!(*.test).js'` .  IMO the filter should be applied for the absolute path of `src/index.js` - the importer.

Fortunately, the include/exclude options accept both glob or regexp, so it can still be used with a regexp. I don't see a way how to match a glob with another glob :wink: 

Nevertheless after a moment of confusion, the current implementation actually makes it easier for my case.

Is the current usage of filter (include, exclude) intentional?
If so, there is no problem, but it could probably be mentioned in the docs/readme.
If not and you think it should be fixed,  pls keep also an option to filter import-globs by regexp.

**Comment by kei-ito** [comment]
The include/exclude option specifies a range of files in which this plugin works. It is useful when this plugin conflicts with other plugins.

The option is not intended to filter glob results. And features distorting glob results won't be implemented because it changes behaviors of program without modifying codes.

If you have a problem with glob results, the glob pattern might be wrong.

Anyway, I noticed that the documentation is unfriendly, so I will update it later. Thank you.

**Comment by kei-ito** [comment]
If you have a problem with glob results, it should be resolved by modifying the glob pattern. If it is impossible, A new import syntax is required. I'll try it in the next release.

**Comment by stropho** [comment]
So I think we have the same opinion about include/exclude. But maybe I misunderstood. I don't really have an issue with the plugin. I just think the behavior is not correct.

> The option is not intended to filter glob results

Yup, I agree, I don't want that. That's what the glob is for. I could possible do that in `intercept`

> The include/exclude option specifies a range of files in which this plugin works.

Agree, but that is **NOT** how it works now
```js
if (!filter(importee) || !importee.includes('*')) {
                return null;
}
```
it checks the `importee` glob string , or other imported paths.

**Comment by kei-ito** [comment]
Ah, I got it. If the include option is set, glob patterns will be ignored unless the include patterns matches the glob patterns. It is a bug.

**Comment by kei-ito** [comment]
I added the test ↓ and [v0.2.4 didn't pass the test](https://travis-ci.com/kei-ito/rollup-plugin-glob-import/builds/136302780).

https://github.com/kei-ito/rollup-plugin-glob-import/blob/dccdad75a570d375ff8d9ed48dfde788d3e74769/test/filter/test.js#L16-L25

https://github.com/kei-ito/rollup-plugin-glob-import/pull/72/commits/f74aedb91b2053f1b98a9949299623482d7ca00f resolved the issue:

```diff
- if (!filter(importee) || !importee.includes('*')) {
+ if (!filter(importer) || !importee.includes('*')) {
```

Thank you for your suggestion.

**Comment by kei-ito** [comment]
I've published v0.4.5. Now the include/exclude works as I expected. Can you try it and close this issue if it works?

**Comment by stropho** [comment]
yup, works as intended
